frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,646 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
+#[proc_macro_attribute]
+pub fn getter(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The optional attribute `#[pallet::storage_prefix = "SomeName"]` allows you to define the
+/// storage prefix to use.
+///
+/// E.g:
+///
+/// ```ignore
+/// #[pallet::storage]
+/// #[pallet::storage_prefix = "foo"]
+/// #[pallet::getter(fn my_storage)]
+/// pub(super) type MyStorage<T> = StorageMap<Hasher = Blake2_128Concat, Key = u32, Value = u32>;
+/// ```
+///
+/// or
+///
+/// ```ignore
+/// #[pallet::storage]
+/// #[pallet::getter(fn my_storage)]
+/// pub(super) type MyStorage<T> = StorageMap<_, Blake2_128Concat, u32, u32>;
+/// ```
+#[proc_macro_attribute]
+pub fn storage_prefix(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The optional attribute `#[pallet::unbounded]` declares the storage as unbounded. When
+/// implementating the storage info (when `#[pallet::generate_storage_info]` is specified on
+/// the pallet struct placeholder), the size of the storage will be declared as unbounded. This
+/// can be useful for storage which can never go into PoV (Proof of Validity).
+#[proc_macro_attribute]
+pub fn unbounded(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The optional attribute `#[pallet::whitelist_storage]` will declare the
+/// storage as whitelisted from benchmarking. Doing so will exclude reads of
+/// that value's storage key from counting towards weight calculations during
+/// benchmarking.
+///
+/// This attribute should only be attached to storages that are known to be
+/// read/used in every block. This will result in a more accurate benchmarking weight.
+///
+/// ### Example
+/// ```ignore
+/// #[pallet::storage]
+/// #[pallet::whitelist_storage]
+/// pub(super) type Number<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+/// ```
+///
+/// NOTE: As with all `pallet::*` attributes, this one _must_ be written as
+/// `#[pallet::whitelist_storage]` and can only be placed inside a `pallet` module in order for
+/// it to work properly.
+#[proc_macro_attribute]
+pub fn whitelist_storage(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The `#[pallet::type_value]` attribute lets you define a struct implementing the
+/// `Get` trait to ease the use of storage types. This attribute can be used
+/// multiple times.
+///
+/// Item must be defined as:
+///
+/// ```ignore
+/// #[pallet::type_value]
+/// fn $MyDefaultName<$some_generic>() -> $default_type $optional_where_clause { $expr }
+/// ```
+///
+/// I.e.: a function definition with generics none or `T: Config` and a returned type.
+///
+/// E.g.:
+///
+/// ```ignore
+/// #[pallet::type_value]
+/// fn MyDefault<T: Config>() -> T::Balance { 3.into() }
+/// ```
+///
+/// NOTE: This attribute is meant to be used alongside [`#[pallet::storage]`](`macro@storage`)
+/// to define some specific default value in storage.
```

Comment:
Suggest bring the note up - it wasn't clear what this was for until I read to the end and found out it was for setting a storage's default value.
```suggestion
/// The `#[pallet::type_value]` attribute lets you define a struct implementing the
/// `Get` trait to ease the use of storage types. This attribute is meant to be used alongside [`#[pallet::storage]`](`macro@storage`)
/// to define a storage's default value. This attribute can be used
/// multiple times.
///
/// Item must be defined as:
///
/// ```ignore
/// #[pallet::type_value]
/// fn $MyDefaultName<$some_generic>() -> $default_type $optional_where_clause { $expr }
/// ```
///
/// I.e.: a function definition with generics none or `T: Config` and a returned type.
///
/// E.g.:
///
/// ```ignore
/// #[pallet::type_value]
/// fn MyDefault<T: Config>() -> T::Balance { 3.into() }
/// ```
```